src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -2139,8 +2140,148 @@ def __repr__(self) -> str:
         """
         Return a string representation for this object.
         """
-        WarningMessage.index_to_pandas_warning("__repr__")
-        return self.to_pandas().__repr__()
+        # Create the representation for each field in the index and then join them.
+        # First, create the data representation.
+        # When the number of elements in the Index is greater than the number of
+        # elements to display, display only the first and last 10 elements.
+        max_seq_items = native_pd.get_option("display.max_seq_items") or 100
+        length_of_index, _, temp_df = self.to_series()._query_compiler.build_repr_df(
+            max_seq_items, 1
+        )
+        if isinstance(temp_df, native_pd.DataFrame) and not temp_df.empty:
+            local_index = temp_df.iloc[:, 0].to_list()
```

Comment:
If you're materializing the data locally, why not just call `repr(native_pd.Index(local_index))`?

Reply:
I guess I could do that? But wouldn't that pull all of the data into local memory?

Reply:
So, `build_repr_df` uses a single query to retrieve the length of the index and also only pick out as many elements needed to be displayed -- if too long, I only need the first and last ten elements. Otherwise, I take only the first 200. If I call `repr(native_pd.Index(local_index))`, it won't display the results correctly since only some of the elements will be present in `local_index`.